src/Tabs/TabGroup.js

```diff
@@ -0,0 +1,111 @@
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import { TabContent } from './_TabContent';
+import React, { Component } from 'react';
+
+export class TabGroup extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedId: props.selectedId
+        };
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        return nextProps.selectedId !== prevState.selectedId ? { selectedId: nextProps.selectedId } : null;
+    }
+
+    // set selected tab
+    handleTabSelection = (event, id) => {
+        event.preventDefault();
+        this.setState({
+            selectedId: id
+        });
+
+        if (this.props.onTabClick) {
+            this.props.onTabClick(event, id);
```

Comment:
TIP: If you give `onTabClick` a default value of an empty function, you won't need to do the "if exists" check here.